src/cryptonote_basic/cryptonote_boost_serialization.h

```diff
@@ -339,6 +340,37 @@ namespace boost
     if (x.type == rct::RCTTypeBulletproof || x.type == rct::RCTTypeBulletproof2)
       a & x.p.pseudoOuts;
   }
+
+  template <class Archive>
+  inline void serialize(Archive &a, cryptonote::difficulty_type &x, const boost::serialization::version_type ver)
+  {
+    if (Archive::is_loading::value)
+    {
+      // load high part
+      uint64_t v = 0;
+      a & v;
+      x = v;
+      // load low part
+      x = x << 64;
+      a & v;
+      x += v;
+    }
+    else
+    {
+      // store high part
+      cryptonote::difficulty_type x_ = x;
+      x_ = x >> 64;
+      uint64_t v = x_.convert_to<uint64_t>();
+      a & v;
+      // store low part
+      x_ = x;
+      x_ = x_ << 64;
+      x_ = x_ >> 64;
+      v = x_.convert_to<uint64_t>();
```

Comment:
Can't this be simply `v = x.convert_to<uint64_t>();`, as done in src/serialization/difficulty_type.h?

Reply:
Yes, looks like boost will just truncate.

Reply:
#$%^&$ boost truncates for 512 bit (which I tested with), but not with 128 bits. So I'll keep the shifts.

Reply:
Really? That's quite surprising and confusing :) Then I think the same bitshift should be done in src/serialization/difficulty_type.h as well.

Reply:
Perhaps a noob question, but why not doing a `x_ = x << 64;` in one instruction instead of two for the first shift?

Reply:
No particular reason I think.